mii/models/providers/llm.py

```diff
@@ -101,15 +103,32 @@ def create_checkpoint_dict(model_name, model_path, mii_config):
         return data
 
 
+def _attempt_load(load_fn, model_name, cache_path, kwargs={}):
+    try:
+        value = load_fn(model_name, **kwargs)
+    except OSError:
+        print(f'Attempted load but failed, retrying using cache_dir={cache_path}')
+        value = load_fn(model_name, cache_dir=cache_path, **kwargs)
+    return value
+
+
 # TODO: This function is a hack for the Bloom models and will be replaced with a LargeModel provider code path
 def load_hf_llm(model_path, model_name, task_name, mii_config):
     deepspeed.init_distributed('nccl')
     local_rank = int(os.getenv('LOCAL_RANK', '0'))
     world_size = int(os.getenv('WORLD_SIZE', '1'))
 
-    tokenizer = AutoTokenizer.from_pretrained(model_name)
-    config = AutoConfig.from_pretrained(model_name)
-    with OnDevice(dtype=torch.float16, enabled=True):
+    cache_path = mii_cache_path()
+
+    tokenizer = _attempt_load(AutoTokenizer.from_pretrained,
+                              model_name,
+                              cache_path,
+                              kwargs={"padding_side": 'left'})
+    tokenizer.pad_token = tokenizer.eos_token
+
+    config = _attempt_load(AutoConfig.from_pretrained, model_name, cache_path)
+
+    with OnDevice(dtype=torch.float16, device='meta', enabled=True):
```

Comment:
dtype here should probably follow the model dtype and not be hardcoded to float16

Reply:
moved to other PR